### ISSUE TITLE
COM812 and COM819 do not support type parameter lists

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#18844]
### Summary

[`missing-trailing-comma` (COM812)](https://docs.astral.sh/ruff/rules/missing-trailing-comma/) and [`prohibited-trailing-comma` (COM819)](https://docs.astral.sh/ruff/rules/prohibited-trailing-comma/) have false negatives for type parameter lists, wherein trailing commas make no difference. The comma rules for type parameter lists should be the same as for normal parameter lists.

COM812 should fix this:
```python
type X[
    T
] = T
def f[
    T
](): pass
class C[
    T
]: pass
```
to this:
```python
type X[
    T,
] = T
def f[
    T,
](): pass
class C[
    T,
]: pass
```

COM819 should fix this:
```python
type X[T,] = T
def f[T,](): pass
class C[T,]: pass
```
to this:
```python
type X[T] = T
def f[T](): pass
class C[T]: pass
```

### Version

ruff 0.12.0 (87f0feb21 2025-06-17)
